crates/hyperqueue/src/server/state.rs

```diff
@@ -264,6 +264,12 @@ impl State {
         for task_id in msg.running_tasks {
             let job = self.get_job_mut_by_tako_task_id(task_id).unwrap();
             job.set_waiting_state(task_id);
+
+            // Increment crash counter for task and check
+            job.set_crashed_task(task_id);
+            if job.is_crashed_task(task_id) {
+                job.set_cancel_state(task_id, backend);
```

Comment:
@spirali is it enough to call `set_cancel_state` here?

Reply:
No, it will just set the internal booking of HyperQueue but not inform Tako.
You should probably call ``cancel_tasks_from_callback`` from here.